text/0401-coercions.md

```diff
@@ -321,18 +321,26 @@ following holds:
 
 * `e` has type `T` and `T` coerces to `U`;
 
-* `e` has type `*T` and `U` is `*U_0` (i.e., between any raw pointers);
+* `e` has type `*T`, `U` is `*U_0`, and either `U_0: Sized` or
+    unsize_kind(`T`) = unsize_kind(`U_0`);
```

Comment:
My point is that `as` is one of the most overloaded entities in the language and now you are throwing even more semantics into it without sufficient reasons. Ideally even `u32 -> u8` would be done with a specialized method, like `truncate()` and not with omnipotent `as`.